Mathlib/Tactic/Linter/DirectoryDependency.lean

```diff
@@ -472,25 +547,75 @@ end DirectoryDependency
 
 open DirectoryDependency
 
-@[inherit_doc Mathlib.Linter.linter.directoryDependency]
-def directoryDependencyCheck (mainModule : Name) : CommandElabM (Option MessageData) := do
-  unless Linter.getLinterValue linter.directoryDependency (← getOptions) do
-    return none
-  let env ← getEnv
-  let imports := env.allImportedModuleNames
+#guard Lean.Name.isPrefixOf `Mathlib.Util `Mathlib.Util.Basic == true
+
+/-- Check if one of the imports `imports` to `mainModule` is forbidden by `forbiddenImportDirs`;
+if so, return an error describing how the import transitively arises. -/
+def _checkBlocklist (env : Environment) (mainModule : Name) (imports : Array Name) : Option MessageData := Id.run do
```

Comment:
Why does this definition have an underscore? (Why not `private def`?)

Reply:
I've been programming too much Python lately :-) A private def is fine.